### ISSUE TITLE
[Security] User checker in new security system

### ISSUE THREAD

**Issue by lugosium** [symfony/symfony-docs#13873]
Hello,

I try to switch to the new security system to test it, but I don’t know how to migrate my current user checker, it seems that we cannot use them anymore. 

Do I need to write a custom authenticator? It seems overkill for adding pre auth check. 
And I cannot extend the form login authenticator because of abstract services. 

I certainly miss something and look around but cannot find what. 

Sorry if i missed something or misunderstood the new system.

**Comment by wouterj** [comment]
The new system had a bug with custom user checkers, see https://github.com/symfony/symfony/issues/37365 for a quick fix (the good fix will be released with 5.1.3).

You can also create a new [event listener](https://symfony.com/doc/current/event_dispatcher.html) on the `CheckPassportEvent` event and then execute your custom user checks:

```php
// ...
use Symfony\Component\EventDispatcher\EventSubscriberInterface;
use Symfony\Component\Security\Http\Event\CheckPassportEvent;
use Symfony\Component\Security\Http\Authenticator\Passport\UserPassportInterface;

class CustomUserChecksListener implements EventSubscriberInterface
{
    public static function getSubscribedEvents()
    {
        return [CheckPassportEvent::class => 'onCheckUser'];
    }

    public function onCheckUser(CheckPassportEvent $event)
    {
        if (!$event->getPassport() instanceof UserPassportInterface) {
            return;
        }

        $user = $event->getPassport()->getUser();
        // ... do checks on $user and throw exception in case of errors
    }
}
```

**Comment by lugosium** [comment]
Thank you @wouterj for your quick answer!

I asked on the slack channel but do not get any answer and i thought that maybe the documentation lack on this part, user checker, and could help other developers if we add an explanation for this case, it was the goal of this issue. But if it's an issue, indeed, no need for this.

I didn't find the existing ticket, sorry!

Thank you for the new security system, it looks really great and everything work fine for the moment 👏
